ui/plugin-system/src/runtime/time-series-queries.ts

```diff
@@ -12,37 +12,27 @@
 // limitations under the License.
 
 import { TimeSeriesQueryDefinition } from '@perses-dev/core';
-import { useQuery } from '@tanstack/react-query';
+import { useQuery, useQueries } from '@tanstack/react-query';
 import { TimeSeriesQueryContext } from '../model';
+import { usePluginRegistry } from '../components';
 import { useTemplateVariableValues } from './template-variables';
 import { useTimeRange } from './time-range';
 import { usePlugin } from './plugins';
 import { useDatasourceStore } from './datasources';
 
-type UseTimeSeriesQueryOptions = {
+export interface UseTimeSeriesQueryOptions {
   suggestedStepMs?: number;
-};
+}
 
 /**
- * Runs a Graph Query using a plugin and returns the results.
+ * Runs a time series query using a plugin and returns the results.
  */
-export const useTimeSeriesQueryData = (definition: TimeSeriesQueryDefinition, options?: UseTimeSeriesQueryOptions) => {
+export const useTimeSeriesQuery = (definition: TimeSeriesQueryDefinition, options?: UseTimeSeriesQueryOptions) => {
   const { data: plugin } = usePlugin('TimeSeriesQuery', definition.spec.plugin.kind);
-
-  // Build the context object from data available at runtime
-  const { timeRange } = useTimeRange();
-  const variableState = useTemplateVariableValues();
-  const datasourceStore = useDatasourceStore();
-
-  const context: TimeSeriesQueryContext = {
-    suggestedStepMs: options?.suggestedStepMs,
-    timeRange,
-    variableState,
-    datasourceStore,
-  };
+  const context = useTimeSeriesQueryContext(options);
 
   const key = [definition, context] as const;
-  const { data, isLoading, error } = useQuery(
+  return useQuery(
```

Comment:
Are we committing to the react-query interface being tightly coupled with our plugin system interface? I'm not against it, more of a general question. I personally love it but wonder if we'll regret that in 2 years lol

Reply:
Good question--this was kinda a quick fix, but maybe I should be more selective about the return types to hide `react-query` a bit? Was definitely something I was thinking about when I did it. With all the generics involved in the return type, I basically punted. 😂

Reply:
Yeah maybe selecting a few of the properties like `data` `error` `isFetching` but beats me 🤷 . Their API is so good that I want all the properties haha